server/controllers/stock/core.js

```diff
@@ -1010,12 +1019,22 @@ async function getInventoryMovements(params) {
       const newQuantity = line.quantity + stockQuantity;
       // fix negative value disorder
       // ignoring negative stock value by setting them to movement value for exit
-      const newValue = (stockValue < 0)
-        ? (line.unit_cost * line.quantity)
-        : (line.unit_cost * line.quantity) + stockValue;
+      let newValue;
+      let newCost;
+      if (isCostRegulatorDepot) {
```

Comment:
I'm not sure the cost regulator depot idea makes sense.  Can't we just use the stock entries from only DONATION, PURCHASE, and INTEGRATION to get the value?  Why do we need to consider the depot?

Reply:
The reason for that is restrict calculation of WAC to certain depots (WAC calculation is only based on entries PURCHASE, DONATION, INTEGRATION); if remove the depot filter (is_cost_regulator), in case a less important depot perform a stock entry (such as INTEGRATION) with bad cost it will affect the entire enterprise WAC value.

Reply:
I think I understand, but I'm not sure I agree.  Shouldn't the value of the stock from integration at any depot affect the cost of stock?  It seems like we'll just ignore certain depots and let users put whatever they want as the value of stock in those depots when it isn't true!